testing/integration/ess/beat_receivers_test.go

```diff
@@ -737,9 +735,18 @@ agent.monitoring.enabled: false
 
 	// use a subcontext for the agent
 	agentProcessCtx, agentProcessCancel := context.WithCancel(ctx)
-	fixture, cmd, output := prepareAgentCmd(t, agentProcessCtx, processConfig)
+	fixture, cmd, processOutput := prepareAgentCmd(t, agentProcessCtx, processConfig)
 
 	require.NoError(t, cmd.Start())
+	fixture.RegisterCleanup()
```

Comment:
Instead of registering only the cleanup functions here,  is it not possibleto use the normal Install() and  run with/without beats receivers as subtests and then compare in the parent test ? Are install logs compared as well ?

Reply:
Installing makes agent run as a service, which makes fetching logs locally more involved, especially given the test runs on all three operating systems. I think the right thing to do here is probably to just spin up a stack and query the logs from ES based on agent id. That feels like overkill, but it would make the test much simpler.

Reply:
I rewrote this test using standard integration test fixtures in https://github.com/elastic/elastic-agent/pull/10153. I'm going to close this PR. Thanks for the hint, Paolo!